### ISSUE TITLE
Add option for consolidating all constructors/destructors into their expected symbols

### ISSUE THREAD

**Issue by khuey** [gimli-rs/cpp_demangle#91]
A debugger does not want to print `MyClass::base object constructor`, it wants to print `MyClass::MyClass`.

I have no idea how to do this though :)  Pointers appreciated.

**Comment by fitzgen** [comment]
Is this what `c++filt` and `libiberty` do?

We could probably add some information to `DemangleContext` or (if lifetime issues force us) another parameter to the `Demangle` trait.

**Comment by khuey** [comment]
```
khuey@minbar:~/dev/cpp_demangle$ ./target/debug/cppfilt _ZN6ParentC2Eh
Parent::base object constructor(unsigned char)
khuey@minbar:~/dev/cpp_demangle$ c++filt _ZN6ParentC2Eh
Parent::Parent(unsigned char)
```

**Comment by fitzgen** [comment]
I'm working on this now as its the next thing blocking bumping the libiberty threshold.

**Comment by fitzgen** [comment]
Fixed in #127
